silkworm/db/datastore/snapshots/history_range_by_keys_query.hpp

```diff
@@ -94,7 +94,7 @@ struct HistoryRangeByKeysSegmentQuery {
 
     auto exec(Bytes key_start, Bytes key_end, datastore::Timestamp timestamp, bool ascending) {
         auto exec_func = [query = *this, key_start = std::move(key_start), key_end = std::move(key_end), timestamp, ascending]() mutable {
-            return query.exec_with_eager_begin(std::move(key_start), std::move(key_end), timestamp, ascending);
+            return query.exec_with_eager_begin(key_start, std::move(key_end), timestamp, ascending);
```

Comment:
If you decide to go with `const Bytes& key_start` I'd still move it for symmetry as you do in other 2 places unless it produces a warning (in which case need to fix 2 other places).